e2e/example_test.go

```diff
@@ -0,0 +1,72 @@
+package e2e
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"gotest.tools/assert"
+
+	"github.com/docker/app/internal"
+
+	"gotest.tools/icmd"
+
+	_ "github.com/docker/cli/cli/compose/schema/data"
+)
+
+const (
+	metadataJsonSchemaVersion = "v0.2"
+	metadataJsonSchema        = "specification/schemas/metadata_schema_" + metadataJsonSchemaVersion + ".json"
+	composeJsonSchemaVersion  = "v3.6"
+	composeJsonSchema         = "vendor/github.com/docker/cli/cli/compose/schema/data/config_schema_" + composeJsonSchemaVersion + ".json"
+)
+
+func TestExamplesAreValid(t *testing.T) {
+	filepath.Walk("../examples", func(p string, info os.FileInfo, err error) error {
+		if !strings.HasSuffix(path.Base(p), internal.AppExtension) {
+			return nil
+		}
+		if info.IsDir() {
+			validateMetadata(t, path.Join(p, internal.MetadataFileName))
+			validateRenderedComposeFile(t, p)
+		} else {
+			validateMetadata(t, p)
+			validateRenderedComposeFile(t, p)
+		}
+		return filepath.SkipDir
+	})
+}
+
+func validateMetadata(t *testing.T, p string) {
+	data, err := ioutil.ReadFile(p)
+	assert.NilError(t, err)
+	validateYaml(t, data, "../"+metadataJsonSchema, p)
```

Comment:
Shouldn't that be `filepath.Join("..",metadataJsonSchema)`?

Reply:
oh... right 😅

Reply:
👍